File: lib/server/accounts.ts
const exp = exports.ox_core.CAccount;

export class CAccount {
  owner: number | string;

  constructor(data: any) {
    this.owner = data.owner;
  }

  get(key: string): any {
    return exp(this.owner, "get", key);
  }

  set(key: string, value: number) {
    exp(this.owner, "set", key, value);
  }

  add(key: string, value: number) {
    exp(this.owner, "add", key, value);
  }

  remove(key: string, value: number) {
    exp(this.owner, "remove", key, value);
  }
}

export function GetAccounts(owner: number | string) {
  return new CAccount({ owner: owner });
}

File: imports/server/accounts.lua
local ox_core = exports.ox_core
local CAccount = {}
local AccountExports = {}
setmetatable(AccountExports, {
    __index = function(_, index)
        AccountExports = Ox.AccountExports()
        return AccountExports[index]
    end
})

function CAccount:__index(index)
    local export = AccountExports[index]

    if export then
        return function(...)
            return ox_core:CAccount(self.owner, index, ...)
        end
    end
end

function Ox.GetAccounts(owner)
    return setmetatable({
        owner = owner
    }, CAccount)
end
